src/pdf_beamtime/src/simple_server.cpp

```diff
@@ -0,0 +1,182 @@
+#include <pdf_beamtime/simple_server.hpp>
+
+using moveit::planning_interface::MoveGroupInterface;
+
+SimpleServer::SimpleServer(
+  const std::string & move_group_name = "ur_manipulator",
+  const rclcpp::NodeOptions & options = rclcpp::NodeOptions())
+: node_(std::make_shared<rclcpp::Node>("simple_server", options))
+{
+  // Add the obstacles
+  this->planning_scene_interface = new moveit::planning_interface::PlanningSceneInterface();
+  this->planning_scene_interface->applyCollisionObjects(create_env());
+
+  this->move_group_interface_ = new MoveGroupInterface(this->node_, move_group_name);
+
+  // Create the action server
+  this->action_server_ = rclcpp_action::create_server<SimpleActionMsg>(
+    this->node_,
+    this->ACTION_NAME,
+    std::bind(&SimpleServer::handle_goal, this, std::placeholders::_1, std::placeholders::_2),
+    std::bind(&SimpleServer::handle_cancel, this, std::placeholders::_1),
+    std::bind(&SimpleServer::handle_accepted, this, std::placeholders::_1));
+
+}
+rclcpp::node_interfaces::NodeBaseInterface::SharedPtr SimpleServer::getNodeBaseInterface()
+// Expose the node base interface so that the node can be added to a component manager.
+{
+  return node_->get_node_base_interface();
+}
+
+rclcpp_action::GoalResponse SimpleServer::handle_goal(
+  const rclcpp_action::GoalUUID & uuid,
+  std::shared_ptr<const SimpleActionMsg::Goal> goal)
+{
+  (void)uuid;
+  return rclcpp_action::GoalResponse::ACCEPT_AND_EXECUTE;
+}
+
+void SimpleServer::handle_accepted(
+  const std::shared_ptr<rclcpp_action::ServerGoalHandle<SimpleActionMsg>> goal_handle)
+{
+  using namespace std::placeholders;
+  // this needs to return quickly to avoid blocking the executor, so spin up a new thread
+  std::thread{std::bind(&SimpleServer::execute, this, _1), goal_handle}.detach();
+}
+
+// Receiving the cancel request.
+rclcpp_action::CancelResponse SimpleServer::handle_cancel(
+  const std::shared_ptr<rclcpp_action::ServerGoalHandle<SimpleActionMsg>> goal_handle)
+{
+  RCLCPP_INFO(node_->get_logger(), "Received request to cancel goal");
+  (void)goal_handle;
+  return rclcpp_action::CancelResponse::ACCEPT;
+}
+
+void SimpleServer::execute(
+  const std::shared_ptr<rclcpp_action::ServerGoalHandle<SimpleActionMsg>> goal_handle)
+{
+  // A cool Moveit function
+  std::vector<double> home_angles = node_->get_parameter("home_angles").as_double_array();
+
+  this->move_group_interface_->setJointValueTarget(home_angles);
```

Comment:
If [when] you change the move_group_interface to an object not a pointer, the syntax here will chage.

Reply:
https://github.com/maffettone/erobs/blob/9f696b7d2d205fd2038f6a764db896805352b3fb/src/pdf_beamtime/src/pdf_beamtime_server.cpp#L62
-> changed to .